### ISSUE TITLE
Negative sampling strategy

### ISSUE THREAD

**Issue by weihangzhang** [amzn/ss-aga-kgc#10]
Hi,

I have noticed that the negative sampling strategy employed is as below:

`    

    def get_negative_samples(self,batch_size_each):
        rand_negs = torch.randint(high=self.num_entity, size=(batch_size_each,),
                                  device=self.device)  # [b,num_neg = 1]
  
        rand_negs = rand_negs.view(-1,1)
        
        return rand_negs
 `

Please correct me if I am wrong: 
It seems like random negative tails are smapled uniformly from the total entity set. However, shouldn't we also make sure to not sample from the true triples in the training set? For example, in the case where triples (h, r, t1) and (h, r, t2) are both in the training set, the negative sampling of (h, r, t1) could potentially label (h, r, t2) as a negative sample and therefore harm the performance of the model. 

I am curious to hear your thoughts on this. Please feel free to let me know if there is anything I am missing here.

**Comment by ZijieH** [comment]
Hi Weihang,
    That is correct since it is possible for random sampling to sample the true tail for a triple. I keep the original implementation from here (https://github.com/stasl0217/KEnS) as the number of entities is relatively large so the possibility would be small. You can of course correct it by excluding true-tails when doing the sampling :)

**Comment by weihangzhang** [comment]
Hi Zijie,

Thanks so much for the prompt reply and the confirmation. Really appreciate it!
